### PR TITLE
Tweak host handling

### PR DESCRIPTION
**Pull request by luqmana**
I tried to tweak the host handling stuff a little to make it a bit more resilient in what it accepts.

### CODE REVIEW

**Comment by jessfraz** [comment]
thank you! this loooks great ill just confirm the tests pass then merge!

**Review by jessfraz — APPROVED** [review]

File: src/docs_man.rs [hunk 1]
```diff
@@ -103,7 +103,7 @@ fn app_has_subcommands(app: &clap::Command) -> bool {
 }
 
 fn subcommand_heading(app: &clap::Command) -> String {
-    match app.get_subommand_help_heading() {
```
Comment (jessfraz) [inline src/docs_man.rs:106]: thank tou for this, i just now realized the difference between the two, thought I was going insane!

**Comment by jessfraz** [comment]
looks like there is a test failure

**Comment by luqmana** [comment]
> looks like there is a test failure

Yea, I can't reproduce on my illumos machine so trying to repro the workflow setup.

File: src/docs_man.rs (hunk 1 above)
Comment (luqmana) [reply src/docs_man.rs:106]: Yea, It took me a bit to realize what was going on.

**Comment by luqmana** [comment]
Ok, the auth tests were failing because `OXIDE_TEST_HOST` wasn't explicity http.

The main test is a bit more perplexing. I was able to repro locally on an Ubuntu box. Basically, the sled-agent docker container was just stuck in a restart loop because of a missing dependency:
```
sled-agent-sim: error while loading shared libraries: libsqlite3.so.0: cannot open shared object file: No such file or directory
```

As a quick test, I just copied those over from my host system and the sled-agent-sim started up fine and passed the tests. I don't quite understand how any of the previous runs passed.

**Comment by luqmana** [comment]
Ok, so definitely not cause by this change. I kicked off the test manually on `main` and it also failed https://github.com/oxidecomputer/cli/runs/5731158437?check_suite_focus=true

**Comment by jessfraz** [comment]
Looks good thank you! And thanks for making the sqlite patch was nice to wake up to since I was planning on doing it!

**Comment by jessfraz** [comment]
Oh weird I guess something went into omicron to make the docker image wonky I’ll pin it to an older version

**Pull request merged**